From 075b85e59d6da98251a2f632751f322bdc16b403 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@st.com>
Date: Mon, 2 Dec 2019 09:26:12 +0100
Subject: [PATCH 154/163] ARM: dts: stm32: fix mdma1 clients channel priority
 level on stm32mp157c

Follow the fix of stm32-mdma bindings regarding channel priority level:
update mdma1 clients channel priority level for those using wrong values.

Change-Id: I99b676cc111288d3214e890b318354252a0a0171
Fixes: 8ecf910a4de8 ("ARM: dts: stm32: Add MDMA support on STM32MP157C")
Upstream-Status: Denied Downstream stm32-gh v5.4-stm32mp [0bc50ee3098641458a2e9a9f035f4e2cf3930ee6]
Signed-off-by: Amelie Delaunay <amelie.delaunay@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/151524
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: CIBUILD <smet-aci-builds@lists.codex.cro.st.com>
Reviewed-by: Jean Philippe ROMAIN <jean-philippe.romain@st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@st.com>
Tested-by: Fabrice GASNIER <fabrice.gasnier@st.com>
---
 arch/arm/boot/dts/stm32mp151.dtsi | 38 +++++++++++++++----------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp151.dtsi b/arch/arm/boot/dts/stm32mp151.dtsi
index 919267b29e8d..871f53ad374a 100644
--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -966,14 +966,14 @@
 			#dma-cells = <4>;
 			st,mem2mem;
 			dma-requests = <8>;
-			dmas = <&mdma1 0 0x11 0x1200000a 0x48000008 0x00000020 1>,
-			       <&mdma1 1 0x11 0x1200000a 0x48000008 0x00000800 1>,
-			       <&mdma1 2 0x11 0x1200000a 0x48000008 0x00200000 1>,
-			       <&mdma1 3 0x11 0x1200000a 0x48000008 0x08000000 1>,
-			       <&mdma1 4 0x11 0x1200000a 0x4800000C 0x00000020 1>,
-			       <&mdma1 5 0x11 0x1200000a 0x4800000C 0x00000800 1>,
-			       <&mdma1 6 0x11 0x1200000a 0x4800000C 0x00200000 1>,
-			       <&mdma1 7 0x11 0x1200000a 0x4800000C 0x08000000 1>;
+			dmas = <&mdma1 0 0x3 0x1200000a 0x48000008 0x00000020 1>,
+			       <&mdma1 1 0x3 0x1200000a 0x48000008 0x00000800 1>,
+			       <&mdma1 2 0x3 0x1200000a 0x48000008 0x00200000 1>,
+			       <&mdma1 3 0x3 0x1200000a 0x48000008 0x08000000 1>,
+			       <&mdma1 4 0x3 0x1200000a 0x4800000C 0x00000020 1>,
+			       <&mdma1 5 0x3 0x1200000a 0x4800000C 0x00000800 1>,
+			       <&mdma1 6 0x3 0x1200000a 0x4800000C 0x00200000 1>,
+			       <&mdma1 7 0x3 0x1200000a 0x4800000C 0x08000000 1>;
 			dma-names = "ch0", "ch1", "ch2", "ch3", "ch4", "ch5", "ch6", "ch7";
 		};
 
@@ -993,14 +993,14 @@
 			#dma-cells = <4>;
 			st,mem2mem;
 			dma-requests = <8>;
-			dmas = <&mdma1 8 0x11 0x1200000a 0x48001008 0x00000020 1>,
-			       <&mdma1 9 0x11 0x1200000a 0x48001008 0x00000800 1>,
-			       <&mdma1 10 0x11 0x1200000a 0x48001008 0x00200000 1>,
-			       <&mdma1 11 0x11 0x1200000a 0x48001008 0x08000000 1>,
-			       <&mdma1 12 0x11 0x1200000a 0x4800100C 0x00000020 1>,
-			       <&mdma1 13 0x11 0x1200000a 0x4800100C 0x00000800 1>,
-			       <&mdma1 14 0x11 0x1200000a 0x4800100C 0x00200000 1>,
-			       <&mdma1 15 0x11 0x1200000a 0x4800100C 0x08000000 1>;
+			dmas = <&mdma1 8 0x3 0x1200000a 0x48001008 0x00000020 1>,
+			       <&mdma1 9 0x3 0x1200000a 0x48001008 0x00000800 1>,
+			       <&mdma1 10 0x3 0x1200000a 0x48001008 0x00200000 1>,
+			       <&mdma1 11 0x3 0x1200000a 0x48001008 0x08000000 1>,
+			       <&mdma1 12 0x3 0x1200000a 0x4800100C 0x00000020 1>,
+			       <&mdma1 13 0x3 0x1200000a 0x4800100C 0x00000800 1>,
+			       <&mdma1 14 0x3 0x1200000a 0x4800100C 0x00200000 1>,
+			       <&mdma1 15 0x3 0x1200000a 0x4800100C 0x08000000 1>;
 			dma-names = "ch0", "ch1", "ch2", "ch3", "ch4", "ch5", "ch6", "ch7";
 		};
 
@@ -1291,7 +1291,7 @@
 			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&rcc HASH1>;
 			resets = <&rcc HASH1_R>;
-			dmas = <&mdma1 31 0x10 0x1000A02 0x0 0x0 0x0 0x0>;
+			dmas = <&mdma1 31 0x2 0x1000A02 0x0 0x0 0x0>;
 			dma-names = "in";
 			dma-maxburst = <2>;
 			status = "disabled";
@@ -1355,8 +1355,8 @@
 			reg = <0x58003000 0x1000>, <0x70000000 0x10000000>;
 			reg-names = "qspi", "qspi_mm";
 			interrupts = <GIC_SPI 92 IRQ_TYPE_LEVEL_HIGH>;
-			dmas = <&mdma1 22 0x10 0x100002 0x0 0x0>,
-			       <&mdma1 22 0x10 0x100008 0x0 0x0>;
+			dmas = <&mdma1 22 0x2 0x100002 0x0 0x0 0x0>,
+			       <&mdma1 22 0x2 0x100008 0x0 0x0 0x0>;
 			dma-names = "tx", "rx";
 			clocks = <&rcc QSPI_K>;
 			resets = <&rcc QSPI_R>;
-- 
2.27.0

